hail/src/main/scala/is/hail/cxx/Emit.scala

```diff
@@ -1012,6 +1014,24 @@ class Emitter(fb: FunctionBuilder, nSpecialArgs: Int, ctx: SparkFunctionContext)
              |})
              |""".stripMargin)
 
+      case ir.NDArrayWrite(nd, path) =>
+        val tub = fb.translationUnitBuilder()
+        tub.include("hail/NDArray.h")
+        val ndt = emit(nd)
+
+        val nativeEncoderClass = CodecSpec.defaultUncompressed.buildNativeEncoderClass(nd.pType, tub)
+        present(
```

Comment:
The way we've handled void types in the jvm bytecode emission is to put all the logic in `setup`; I'd kind of prefer to keep that here since I think it makes more sense. (And if we check for void types in the wrapper function, we don't need to return a placeholder value here, either)